src/Materials/babylon.material.ts

```diff
@@ -788,8 +794,11 @@ module BABYLON {
             this._useUBO = this.getScene().getEngine().supportsUniformBuffers;
 
             if (!doNotAdd) {
-                this._scene.materials.push(this);
-                this._scene.onNewMaterialAddedObservable.notifyObservers(this);
+                this._scene.addMaterial(this);
+            }
+
+            if (scene.useMaterialMeshMap) {

```

Comment:
I fixed this one as scene should be this._scene

Reply:
yes you're right, good catch !